catalog/spatial/csw/spatial-csw-endpoint/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/endpoint/CswEndpoint.java

```diff
@@ -929,10 +946,18 @@ private CswRecordCollection queryCsw(GetRecordsType request) throws CswException
             try {
                 queryRequest = queryFactory.updateQueryRequestTags(queryRequest,
                         request.getOutputSchema());
-                LOGGER.debug("Attempting to execute query: {}", queryRequest);
-                QueryResponse queryResponse = framework.query(queryRequest);
+                queryRequest = setQueryPageSize(queryRequest);
```

Comment:
I believe we have lost the requests original `maxRecords` at this point - which means we will return all results possible even if they have only requested 5 (or any subset).